### ISSUE TITLE
fix: ensure custom name is used in APKG download when converting via Notion integration

### ISSUE THREAD

**Issue by aalemayhu** [2anki/server#763]
Currently we only support this for the HTML upload. Would love to have custom names in the APKG file.

**Comment by aalemayhu** [comment]
Reopening after triage: Custom APKG name via Notion integration; no PR linked, conservative reopen. Reviewed by: engineer agent.

**Comment by aalemayhu** [comment]
Verified against current `main` — the custom deck name now flows through the Notion conversion path the same way it does for HTML uploads. Closing as fixed.

What I traced:

1. **Where the name is captured.** `CardOption.deckName` (`src/lib/parser/Settings/CardOption.ts`) reads `input.deckName`, persisted via the per-page settings endpoint (`POST /api/settings/create/:id`) and rendered in the settings UI for the Notion page (`web/src/components/CardOptionsForm/CardOptionsForm.tsx`).
2. **Where it's loaded.** `CreateJobWorkSpaceUseCase` (`src/usecases/jobs/CreateJobWorkSpaceUseCase.ts`) calls `SettingsRepository.load(owner, pageId)` — same shape as the upload path uses for its saved settings.
3. **Where it becomes the filename.** `BuildDeckForJobUseCase` (`src/usecases/jobs/BuildDeckForJobUseCase.ts` L70–78) computes the filename as `settings.deckName || bl.firstPageTitle || id`, runs it through `toText` + `addDeckNameSuffix`, and writes it to `uploads.filename`. `settings.deckName` is also passed in as `parentName` so the in-Anki deck name matches.
4. **Where the download uses it.** `DownloadController.getFile` (`src/controllers/DownloadController.ts` L52–60) reads `uploads.filename` via `DownloadService.getFilename`, runs `getSafeFilename`, and writes it to `Content-Disposition`.

Repro path that confirms the fix:
- Connect Notion → open the page's settings → set "Deck name" to `Organic Chemistry Ch. 4` → save → trigger conversion → the resulting .apkg downloads as `Organic Chemistry Ch. 4.apkg`, not the Notion page title or the page id.

No PR — the wiring is already in place.
